pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/common/PirActionsRunnerFactory.kt

```diff
@@ -17,55 +17,14 @@
 package com.duckduckgo.pir.internal.common
 
 import android.content.Context
-import com.duckduckgo.common.utils.CurrentTimeProvider
-import com.duckduckgo.common.utils.DispatcherProvider
-import com.duckduckgo.js.messaging.api.JsMessageHelper
-import com.duckduckgo.pir.internal.scripts.PirMessagingInterface
-import com.duckduckgo.pir.internal.scripts.RealBrokerActionProcessor
-import com.duckduckgo.pir.internal.store.PirRepository
-import javax.inject.Inject
+import com.duckduckgo.pir.internal.common.PirJob.RunType
+import dagger.assisted.AssistedFactory
 
-class PirActionsRunnerFactory @Inject constructor(
-    private val pirDetachedWebViewProvider: PirDetachedWebViewProvider,
-    private val dispatcherProvider: DispatcherProvider,
-    private val jsMessageHelper: JsMessageHelper,
-    private val currentTimeProvider: CurrentTimeProvider,
-    private val pirRunStateHandler: PirRunStateHandler,
-    private val pirRepository: PirRepository,
-    private val captchaResolver: CaptchaResolver,
-) {
-    /**
-     * Every instance of PirActionsRunner is created with its own instance of [PirMessagingInterface] and [RealBrokerActionProcessor]
-     */
-    fun createInstance(
+@AssistedFactory
```

Comment:
I think you should move the factory inside the `RealPirActionsRunner` class. That's generally how the factory pattern is done, eg.

```kotlin
class RealPirActionsRunner @AssistedInject constructor(...) {
  @AssistedFactory
  interface Factory {
    create(...): RealPirActionsRunner {}
  }
}
```